rules/CodingStyle/Rector/Foreach_/MultiDimensionalArrayToArrayDestructRector.php

```diff
@@ -152,7 +152,7 @@ private function getUsedVariableNamesInForeachTree(Foreach_ $foreach): array
         $variableNodes = $this->nodeFinder->findInstanceOf($foreach, Variable::class);
 
         return array_unique(array_map(
-            fn (Variable $variable): string => (string) $this->getName($variable),
+            fn (Variable $variable): string => (string) $this->getName($foreach),
```

Comment:
this is bug on `RenameParamToMatchTypeRector`,  that cause test error, need use `$variable`:

```suggestion
            fn (Variable $variable): string => (string) $this->getName($variable),
```


Ref issue and PR:

- https://github.com/rectorphp/rector/issues/8693
- https://github.com/rectorphp/rector-src/pull/6046

Reply:
Got it :+1: